Shared/Services/Backups/ScheduledBackupManager.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  ScheduledBackupManager.swift
+//  InternxtDesktop
+//
+//  Created by Patricio Tovar on 10/6/24.
+//
+
+import Foundation
+import Combine
+
+class ScheduledBackupManager {
+    
+    static let shared = ScheduledBackupManager()
+    private var backupTimer: AnyCancellable?
+    private let backupsService: BackupsService = BackupsService()
+    private let userDefaults = UserDefaults.standard
+    private var initialTimer: Timer?
+    private let BACKUP_FREQUENCY_KEY = "INTERNXT_SELECTED_BACKUP_FREQUENCY"
+    private let LAST_BACKUP_TIME_KEY = "INTERNXT_LAST_BACKUP_TIME_KEY"
+    
+    func startBackupTimer(frequency : BackupFrequencyEnum) {
+        
+        userDefaults.set(frequency.rawValue, forKey: BACKUP_FREQUENCY_KEY)
+        backupTimer?.cancel()
+        initialTimer?.invalidate()
+        
+        guard frequency != .manually else { return }
+        
+        let interval = frequency.timeInterval
+        
+        let now = Date()
+        
+        if let lastBackupTime = userDefaults.object(forKey: LAST_BACKUP_TIME_KEY) as? Date {
+            let elapsedTime = now.timeIntervalSince(lastBackupTime)
+            let timeUntilNextBackup = interval - elapsedTime.truncatingRemainder(dividingBy: interval)
+            
+            // Set a one-time timer to perform the next backup
+            initialTimer = Timer.scheduledTimer(withTimeInterval: timeUntilNextBackup, repeats: false) { [weak self] _ in
+                self?.performBackup()
+                self?.startRecurrentBackupTimer(interval: interval)
+            }
+        } else {
+            startRecurrentBackupTimer(interval: interval)
+        }
+        
+        
+    }
+    
+    
+    private func performBackup() {
+        Task {
+            do {
+                await backupsService.loadAllDevices()
+                backupsService.loadFoldersToBackup()
+                try await backupsService.startBackup { progress in}
+                userDefaults.set(Date(), forKey: LAST_BACKUP_TIME_KEY)
+            } catch {
+                
```

Comment:
Do something with the possible error here, something in the UI should be displayed at least an NSAlert

Reply:
Fixed